cellacdc/widgets.py

```diff
@@ -6754,6 +6762,39 @@ def pointsClicked(self, item, points, event):
         print('')
         print('*'*60)
 
+    def showIDs(self, annotate_labels_idxs=None, init=False):
+        if init:
+            self.annotate_labels_idxs = annotate_labels_idxs
+        if self.annotate_labels_idxs is None:
+            return
+        for i, plotItem in enumerate(self.PlotItems):
+            if i not in self.annotate_labels_idxs:
+                continue
+            imageItem = self.ImageItems[i]
+            lab = imageItem.image
+            rp = skimage.measure.regionprops(lab)
+            for obj in rp:
+                yc, xc = obj.centroid
+                textItem = pg.TextItem(
+                    text=str(obj.label), anchor=(0.5,0.5),
+                )
+                textItem.setPos(xc, yc)
+                plotItem.addItem(textItem)
+                self.textItems.append(textItem)
+
+    def clearLabels(self): # This is very slow and causes some errors, need to refactor at some point
```

Comment:
For clearing test we simply set an empty string `textItem.setText('')`. I would not remove items since we can recycle them